graph/preprocessor.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* Alias preprocessor:
+* The set of elements here are meant to process the alias definition portion of task.yaml
+* files. This is done by unmarshaling these elements which will then be added in a hierarchical
+* manner. Note the input must still be valid Yaml.
+*
+* Existing issues: Once the preTask is parsed out and the definitions are resolved, the read in
+* yaml will be processed to include the appropriate aliases, however this will include the previously
+* parsed in definitions which is not as efficient.
+*
+ */
+
+package graph
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+var (
+	errUnknownAlias              = errors.New("unknown Alias")
+	errImproperDirectiveOverload = errors.New("$ directive can only be overwritten by a single character")
+	errImproperKeyName           = errors.New("alias key names only support alphanumeric characters and _ , - , ! characters")
+	directive                    = '$'
+)
+
+// PreTask intermediate step for processing before complete unmarshall
+type PreTask struct {
+	AliasSrc  []*string         `yaml:"alias-src"`
```

Comment:
Be sure to run the names of the yaml through Steve. Bare minimum we don't have dashes anywhere and use camelCase instead so it would be good to change that.

Reply:
I will make sure to do that